lib/finch/http1/pool.ex

```diff
@@ -74,6 +74,56 @@ defmodule Finch.HTTP1.Pool do
     end
   end
 
+  @impl Finch.Pool
+  def async_request(pool, req, opts) do
+    owner = self()
+
+    pid =
+      spawn(fn ->
+        monitor = Process.monitor(owner)
+        request_ref = receive_next_within!(10)
```

Comment:
We would definitely want this value to be much higher. Otherwise a temporarily slow system will start seeing crashes.

```suggestion
        request_ref = receive_next_within!(5_000)
```